src/hyperion/external_interaction/callbacks/xray_centre/ispyb_mapping.py

```diff
@@ -1,101 +1,68 @@
 from __future__ import annotations
 
+import numpy
 from dodal.devices.oav import utils as oav_utils
 
-from hyperion.external_interaction.callbacks.common.ispyb_mapping import GridScanInfo
 from hyperion.external_interaction.ispyb.data_model import (
     DataCollectionGridInfo,
     DataCollectionInfo,
 )
-from hyperion.external_interaction.ispyb.ispyb_dataclass import Orientation
 
 
 def populate_xz_data_collection_info(
-    grid_scan_info: GridScanInfo,
     full_params,
-    ispyb_params,
     detector_params,
 ) -> DataCollectionInfo:
     assert (
         detector_params.omega_start is not None
         and detector_params.run_number is not None
-        and ispyb_params is not None
         and full_params is not None
     ), "StoreGridscanInIspyb failed to get parameters"
     omega_start = detector_params.omega_start + 90
```

Comment:
Should: We should be able to read the omega from the smargon when we do the pin tip detection, can we make an issue for that?

Reply:
Have made #1327 but is pin tip detection the right place or should it be done during `read_hardware_for_ispyb_pre_collection`?

Reply:
It can't be done during `read_hardware_for_ispyb_pre_collection` because we want the omega for each grid. If we read omega at pre_collection we only get the first grid's omega. We would then have to add the 90 deg in the callback again, which I was trying to avoid. Basically I want the logic of what we're doing to be in the plan, the callback should just be blindly funneling it into ispyb